### PR TITLE
Fix OpenBSD build

### PR DESCRIPTION
**Pull request by riataman**
This fixes the build in OpenBSD.

Still retrace won't run when build because it needs to link to pthread. If I manually link it to pthread it works fine. Hopefully @wizzard  can fix the pthread linking.

### CODE REVIEW

**Comment by wizzard** [comment]
Sure, will take a look today.

**Comment by wizzard** [comment]
Added pthread support.
@riataman with your changes all tests are passing on OpenBSD, good job!

File: common.c [hunk 1]
```diff
@@ -756,7 +754,11 @@ retrace_event(struct rtr_event_info *event_info)
 		return;
 
 	old_trace_state = trace_disable();
+
+	old_trace_state = trace_disable();
```
Comment (ronaldtse) [inline common.c]: Am I reading this right, there are two calls (756, 758) of:
``` c
old_trace_state = trace_disable();
```
?

Comment (riataman) [inline common.c]: Yeah, this one had a bunch of merging issues. I need to go back and rework it.

**Comment by riataman** [comment]
This now works in OpenBSD. Builds in FreeBSD but has a lot of problems, would need to sit down an debug them but I'm not sure that's a priority right now.

**Pull request merged**